Dockerfile

```diff
@@ -1,27 +1,30 @@
-FROM ubuntu:xenial
+FROM ubuntu:bionic
 
-MAINTAINER Carsten Byrman <carsten.byrman@nelen-schuurmans.nl>
+LABEL maintainer="carsten.byrman@nelen-schuurmans.nl"
 
 # Get rid of debconf messages like "unable to initialize frontend: Dialog".
 # https://github.com/docker/docker/issues/4032#issuecomment-192327844
 ARG DEBIAN_FRONTEND=noninteractive
 
-# `python bootstrap.py` succeeds on ubuntu:trusty, but fails on ubuntu:xenial
-# with an SSL certificate error. Installing apt-transport-https helps.
-
 RUN apt-get update && apt-get install -y \
-    apt-transport-https \
-    build-essential \
-    python-dev \
-&& rm -rf /var/lib/apt/lists/*
-
-# The -l option is to fix a problem with large user IDs (e.g. 1235227245).
-# https://forums.docker.com/t/run-adduser-seems-to-hang-with-large-uid/27371/3
-# https://github.com/moby/moby/issues/5419
-ARG uid=1000
-ARG gid=1000
-RUN groupadd -g $gid nens && useradd -lm -u $uid -g $gid nens
-
-# Create and set the working directory for many instructions.
+    python3 \
+    python3-venv \
+ && rm -rf /var/lib/apt/lists/*
+
+# This one works fine (en_US.UTF-8 is not present by default).
+ENV LANG=C.UTF-8
+
+# In my experience, mixing apt and pip managed Python packages often
+# causes trouble. Avoid this by using a venv.
+RUN python3 -m venv /venv
+RUN /venv/bin/pip install --upgrade --no-cache-dir pip pip-tools setuptools wheel
+
+# This is the directory that will be watched for file uploads.
+# NB: We are not really running an FTP server in this setup.
+RUN mkdir -p /srv/ftp
+
+# Create and set the working directory for subsequent instructions.
 WORKDIR /code
-USER nens
+
+ADD requirements.txt .
+RUN /venv/bin/pip-sync --pip-args '--no-cache-dir'
```

Comment:
Ik ben nieuw met pip-tools. Kun je me nog iets vertellen over de installatie van watchman zelf? Zeg maar de `pip -e .`, hoe werkt dat? Ik werk in een andere library met een src/ directory, en dan moet je hem expliciet installeren.

Reply:
Je kunt `/venv/bin/pip install -e .` doen. Pip maakt dan ook gewoon een entry_point aan, zodat `venv/bin/watchman` werkt.